drivers/adc/adc_stm32.c

```diff
@@ -236,6 +236,41 @@ static const uint32_t table_samp_time[] = {
 };
 #endif
 
+/* table to define the LL ADC clock prescaler from the DTS */
```

Comment:
We should add a static assert that `config->clock_prescaler < ARR_SIZE(adc_clock_prescaler)`

Reply:
with ASSERT_NO_MSG